packages/core/mesh/network-manager/src/swarm/peer.ts

```diff
@@ -114,7 +115,9 @@ export class Peer {
 
         if (this.connection) {
           // Close our connection and accept remote peer's connection.
-          await this.closeConnection(new Error('Connection displaced by remote initiator.'));
+          this.displacedConnection = this.connection;
+          this.connection = undefined;
+          await this.closeConnection(this.displacedConnection, new Error('Connection displaced by remote initiator.'));
```

Comment:
to retain class open/close semantics is it possible to keep closeConnection as is (i.e., operate on this._connection) and keep track of the displaced connection separately?

Reply:
I can't make out why you're separately holding on to the "displacedConnection" is that just for testing? If so, that might be better to expose as an `Event`.

```
private readonly _connection = new Event<{ state, Connection >});
```

Reply:
yes, it's used mainly for testing atm - i just was thinking we might want to use it somehow during the diagnostics through the devtools or smth